### ISSUE TITLE
iisnode doesnt appear to allow for catching any process exit signals inside node.  

### ISSUE THREAD

**Issue by tony-gutierrez** [Azure/iisnode#61]
I have tried everything I can think of to detect node shutdown on azure webapps (for graceful exits). I can detect shutdown on normal node on windows, but not in Azure. 

Some listeners that never fire:
```
        process.on('SIGINT', handle);
	process.on('SIGTERM', handle);
	process.on('SIGHUP', handle);
	process.on('SIGBREAK', handle);
	process.on('exit', handle);
```

also the older readline method does not work either:
```
         let rl =require("readline")
	.createInterface({
		input: process.stdin,
		// output: process.stdout
	})
	.on("SIGINT", function () {
		console.log('rlsigint');
	})
``` 

Is it not possible to detect node shutdown at all in iisnode for the purposes of cleaning up, sending last logs to db, etc?

**Comment by tony-gutierrez** [comment]
I was working with Azure on this. They had a method where they would create pipes that you could subscribe to for shutdown notification. It worked once and a while barely, and eventually they just stopped responding to my emails.

**Comment by lordbah** [comment]
We're on Windows 10 in case it matters.

**Comment by abondoa** [comment]
@tony-gutierrez @lordbah did you find any way to receive process-exit signals inside the node process?

**Comment by pursual** [comment]
No, the Azure pipes thing never worked, and the people at azure who initially pointed me to it stopped responding.

**Comment by lordbah** [comment]
No, but it became less important because AAG v2 wasn't pulling the same shenanigans. Then I retired and it became _really_ unimportant to me :-)

**Comment by abondoa** [comment]
Alright, thanks for responding so quickly both of you.
